### DISCUSSION TITLE
eCAL 5.10 is coming

### DISCUSSION THREAD

**Discussion by FlorianReimold** [eclipse-ecal/ecal#645] [Announcements]
# eCAL 5.10 is coming!

Next Monday, on 2022-05-09, we are going to release eCAL 5.10. Among the changes are:

- Added **Dark Mode** in all apps *(Activate it by View/Theme/Dark!)*

    [<img src="https://user-images.githubusercontent.com/11774314/166221617-eda26fdf-78e7-4338-ab79-f5897f2458d4.png" width="300"/>](https://user-images.githubusercontent.com/11774314/166221617-eda26fdf-78e7-4338-ab79-f5897f2458d4.png)

- eCAL Mon can now display and **dock plugins** in it's main window, not only in the reflection Dialog
- Added eCAL Mon plugin for **plotting numeric data** in a 2D graph

    [<img src="https://user-images.githubusercontent.com/11774314/166221928-7814c139-803a-41e5-8b53-97136056973e.png" width="300"/>](https://user-images.githubusercontent.com/11774314/166221928-7814c139-803a-41e5-8b53-97136056973e.png)

- Added full **UTF-8 Support** for all Apps on Windows
- Added **Shared Memory improvements** (Multi-Buffering & Zero-Copy)

    [<img src="https://user-images.githubusercontent.com/11774314/166221997-fa183b79-930a-44c7-b3cf-aafacfb98843.png" width="300"/>](https://user-images.githubusercontent.com/11774314/166221997-fa183b79-930a-44c7-b3cf-aafacfb98843.png)

- Added **TCP Transport layer** (Improves Network subscriptions of large messages)

# Upgrade

You should not expect many difficulties while upgrading from eCAL 5.9. We did induce an incompatibility though in the Raw Service API. As that API is barely used by user-code (you usually want to use the Protobuf Service API), you will probably not notice it. But if you do, we got you covered with an extra page on how to solve those issues:
https://continental.github.io/ecal/versions/5.10/api_incompatibilities.html#compatibility-details-5-10-service-api

# Legacy

eCAL 5.10 does not compile on Ubuntu 16.04 any more and therefore we will not publish Ubuntu 16.04 builds in our PPA. You can still get it compiling by disabling the GUI Applications, so the Core is still GCC 5 compatible.

eCAL 5.10 will kick eCAL 5.8 out of support. You can use 5.8 as long as you like, but we won't invest time in bugfixes any more. eCAL 5.9 will continue to receive patches for the next 6 month or so (until 5.11 is released)

**Comment by rex-schilasky** [comment]
Great release ! Looking forward to next Monday. Thank you @FlorianReimold and all the other contributors for all the effort you invest over the last months and weeks to make this release happen.

**Comment by ZhenshengLee** [comment]
Great to see that v5.10 released via apt in ubuntu!

**Reply by FlorianReimold** [reply]
It is! it is now officially released both as [manual download](https://continental.github.io/ecal/_download_archive/download_archive_ecal_5_10_0.html), from the eCAL [Rolling release PPA](https://launchpad.net/~ecal/+archive/ubuntu/ecal-latest) and as [5.10 PPA](https://launchpad.net/~ecal/+archive/ubuntu/ecal-5.10)!
